stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -2476,12 +2479,43 @@ internal constructor(
         channelsIds: List<String>,
         lastSyncAt: Date,
     ): Call<List<ChatEvent>> {
-        return api.getSyncHistory(channelsIds, lastSyncAt)
+        val stringDate = streamDateFormatter.format(lastSyncAt)
+
+        return api.getSyncHistory(channelsIds, stringDate)
             .withPrecondition(scope) {
                 checkSyncHistoryPreconditions(channelsIds, lastSyncAt)
             }
     }
 
+    /**
+     * Returns all events that happened for a list of channels since last sync (while the user was not
+     * connected to the web-socket). LastSyncAt is in the format yyyy-MM-dd'T'HH:mm:ss.SSSSSS'Z'. Use this version of
+     * getSyncHistory when high precision is necessary.
+     *
+     * @param channelsIds The list of channel CIDs. Cannot be empty.
+     * @param lastSyncAt The last time the user was online and in sync. Shouldn't be later than 30 days.
+     *
+     * @return Executable async [Call] responsible for obtaining missing events.
+     */
+    @CheckResult
+    @InternalStreamChatApi
```

Comment:
Does it make sense to make it public? 🤔 Maybe we should also think about deprecating the old `getSynHistory` with `Date`?

Reply:
making it public can be useful for customers that want to use a greater precision. About deprecating the old one... I would say that using both is preferable, because the string version may be considered hard to use if the customer only wants to generate the time locally.